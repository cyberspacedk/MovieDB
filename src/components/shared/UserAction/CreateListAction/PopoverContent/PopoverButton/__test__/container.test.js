import React from 'react';
import { shallow } from 'enzyme';
import PopoverButtonContainer from '../container';

describe('CreateListActionConnected ', () => {
  const props = {
    movieId: 777,
    item: { id: 9 },
    addMovieToList: jest.fn(),
  };

  const container = shallow(<PopoverButtonContainer {...props} />);
  const instance = container.instance();

  it('Should match its snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('Should call method handlerAddMovieToList', () => {
    instance.handlerAddMovieToList();
    expect(instance.props.addMovieToList).toHaveBeenCalledWith(9, 777);
  });
});
