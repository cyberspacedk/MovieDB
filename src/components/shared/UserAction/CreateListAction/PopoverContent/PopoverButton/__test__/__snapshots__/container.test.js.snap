// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateListActionConnected  Should match its snapshot 1`] = `
<PopoverButton
  handlerAddMovieToList={[Function]}
  item={
    Object {
      "id": 9,
    }
  }
/>
`;
